modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/PortletBase.es.js

```diff
@@ -24,18 +22,39 @@ function toElementHelper(elementOrSelector) {
 	return elementOrSelector;
 }
 
-function isString(val) {
-	return typeof val === 'string';
+class LifeCycles {
+	constructor() {
+		this.created();
+
+		this.attached();
+	}
+	dispose() {
+		this.disposeInternal();
+
+		this.detached();
+
+		this.disposed();
+	}
+	attached() {}
+	created() {}
+	detached() {}
+	disposed() {}
+	disposeInternal() {}
```

Comment:
What's your logic behind this addition/change? I'm not questioning it, I'm just interested as to how you came up to this idea and what makes it good. I don't think I've seen something like this before.

Reply:
you know, I think I may have pushed the wrong branch... this isn't supposed to be here